src/main/java/graphql/normalized/NormalizedField.java

```diff
@@ -110,40 +165,30 @@ public NormalizedField transform(Consumer<Builder> builderConsumer) {
         return builder.build();
     }
 
-    public GraphQLObjectType getObjectType() {
-        return objectType;
+    public Set<String> getObjectTypeNames() {
+        return objectTypeNames;
     }
 
+    public GraphQLObjectType getOneObjectType(GraphQLSchema schema) {
+        return (GraphQLObjectType) schema.getType(objectTypeNames.iterator().next());
+    }
+
+
     public String printDetails() {
 
         StringBuilder result = new StringBuilder();
         if (getAlias() != null) {
             result.append(getAlias()).append(": ");
         }
-        return result + objectType.getName() + "." + fieldDefinition.getName() + ": " + simplePrint(fieldDefinition.getType()) +
-                " (conditional: " + this.isConditional + ")";
-    }
-
-    public String print() {
-        StringBuilder result = new StringBuilder();
-        result.append("(");
-        if (getAlias() != null) {
-            result.append(getAlias()).append(":");
-        }
-        return result + objectType.getName() + "." + fieldDefinition.getName() + ")";
+        return result + objectTypeNamesToString() + "." + fieldName;
     }
 
-    public String printFullPath() {
-        StringBuilder result = new StringBuilder();
-        NormalizedField cur = this;
-        while (cur != null) {
-            if (result.length() > 0) {
-                result.insert(0, "/");
-            }
-            result.insert(0, cur.print());
-            cur = cur.getParent();
+    public String objectTypeNamesToString() {
+        if (objectTypeNames.size() == 1) {
+            return objectTypeNames.iterator().next();
```

Comment:
list.get(0) is quicker - index look up  - no allocation of iterator

```
        return (E) elementData[index];

```

Reply:
it is a Set, so no get(0)